crates/core/src/db/relational_db.rs

```diff
@@ -1449,4 +1473,61 @@ mod tests {
 
     //     Ok(())
     // }
+
+    #[test]
+    fn test_replay_corrupted_log() -> ResultTest<()> {
+        let (db, tmp) = make_test_db()?;
+        let ctx = ExecutionContext::default();
+
+        let table_id = db.with_auto_commit(&ctx, |tx| {
+            db.create_table(
+                tx,
+                table(
+                    "Account",
+                    vec![ColumnDef {
+                        is_autoinc: true,
+                        ..column("deposit", AlgebraicType::U64)
+                    }],
+                    vec![],
+                ),
+            )
+        })?;
+
+        fn balance(ctx: &ExecutionContext, db: &RelationalDB, table_id: TableId) -> ResultTest<u64> {
+            let balance = db.with_auto_commit(ctx, |tx| {
+                let mut sum = 0;
+                for row in db.iter(ctx, tx, table_id)? {
+                    sum += row.view().field_as_u64(0, None)?;
+                }
+                Ok::<_, DBError>(sum)
+            })?;
+
+            Ok(balance)
+        }
+
+        // Deposit some coins, creating 10 transactions.
+        for _ in 0..10 {
+            db.with_auto_commit(&ctx, |tx| db.insert(tx, table_id, product![AlgebraicValue::U64(0)]))?;
+        }
+        assert_eq!(55, balance(&ctx, &db, table_id)?);
+
+        // Close the db an pop a byte from the end of the message log.
```

Comment:
It would be nice to also test the "safeguard" condition (log corrupted in the middle), but the default segment size in 1GiB. Punting on making this configurable at this point.

Reply:
Actually, that’s not true but obscured by the various test helpers. Will add a test before landing.